src/image/MaskedImageFitsReader.cc

```diff
@@ -202,32 +202,37 @@ MaskedImage<ImagePixelT, MaskPixelT, VariancePixelT> MaskedImageFitsReader::read
 
     // only read other planes if they're in their normal HDUs
     if (_imageReader.getHdu() == static_cast<int>(Hdu::Image)) {
-        try {
-            mask = std::make_shared<Mask<MaskPixelT>>(
-                _maskReader.read<MaskPixelT>(bbox, origin, conformMasks, allowUnsafe)
-            );
-        } catch (fits::FitsError& e) {
-            if (needAllHdus) {
-                LSST_EXCEPT_ADD(e, "Reading Mask");
-                throw e;
-            }
-            LOGL_WARN(_log, "Mask unreadable (%s); using default", e.what());
-            // By resetting the status we are able to read the next HDU (the variance).
-            _maskReader._fitsFile->status = 0;
-        }
-        try {
-            variance = std::make_shared<Image<VariancePixelT>>(
-                _varianceReader.read<VariancePixelT>(bbox, origin, allowUnsafe)
-            );
-        } catch (fits::FitsError& e) {
-            if (needAllHdus) {
-                LSST_EXCEPT_ADD(e, "Reading Variance");
-                throw e;
-            }
-            LOGL_WARN(_log, "Variance unreadable (%s); using default", e.what());
-            // By resetting the status we are able to read the next HDU (the variance).
-            _varianceReader._fitsFile->status = 0;
+        // Do not even try if we are missing the HDU
+        if (_maskReader._fitsFile) {
+          try {
+              mask = std::make_shared<Mask<MaskPixelT>>(
+                  _maskReader.read<MaskPixelT>(bbox, origin, conformMasks, allowUnsafe)
+              );
+          } catch (fits::FitsError& e) {
+              if (needAllHdus) {
+                  LSST_EXCEPT_ADD(e, "Reading Mask");
+                  throw e;
+              }
+              LOGL_WARN(_log, "Mask unreadable (%s); using default", e.what());
```

Comment:
Looking at this code in isolation, it seems like it'd be appropriate to emit this warning when `_maskReader._fitsFile` is null, too.  But in the higher-level context of the ticket, it seems we load `Exposure` or `MaskedImage` objects from single-plane FITS files all the the time, and don't want warnings when we do that, so maybe the right fix is to remove the warning from both code paths?

In any case, we should probably obey the `needAllHdus` logic when `_maskFitsReader._fitsFile` is null, and throw an exception.  I don't think that's used very often (if at all), but it'd certainly be confusing to pass `needAllHdus=True` and then silently only get the image plane loaded.